frmts/jpegxl/jpegxl.cpp

```diff
@@ -2536,6 +2572,18 @@ GDALDataset *JPEGXLDataset::CreateCopy(const char *pszFilename,
                          "JxlEncoderSetExtraChannelName() failed");
                 return nullptr;
             }
+            if (channelType == JXL_CHANNEL_ALPHA && fAlphaDistance >= 0.0f)
+            {
+#if HAVE_JxlEncoderSetExtraChannelDistance
```

Comment:
Shouldn't this be moved before the `if` ?

Reply:
done